homeassistant/components/husqvarna_automower/__init__.py

```diff
@@ -52,8 +73,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Handle unload of an entry."""
-    coordinator: AutomowerDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
-    await coordinator.shutdown()
+    automower_entry_data: AutomowerEntryData = hass.data[DOMAIN][entry.entry_id]
+    await automower_entry_data.coordinator.shutdown()
```

Comment:
The `api.close` method doesn't seem relevant for Home Assistant to call anymore.

The token task is never set in the library at all.

https://github.com/Thomas55555/aioautomower/blob/3d0cdee7a2aea27e618d378a89280d631c16f238/aioautomower/session.py#L309

The rest task isn't called in our case since we don't instantiate the session with `poll` True.

https://github.com/Thomas55555/aioautomower/blob/3d0cdee7a2aea27e618d378a89280d631c16f238/aioautomower/session.py#L310

Side note: The library should check that there are tasks before gathering the tasks.

https://github.com/Thomas55555/aioautomower/blob/3d0cdee7a2aea27e618d378a89280d631c16f238/aioautomower/session.py#L318